File: src/main/java/com/manageway/domain/entity/user/records/Email.java
package com.manageway.domain.entity.user.records;

import org.springframework.util.Assert;

public record Email(String value) {
    public Email {
        Assert.isNull(value, "Email must be provided.");

        String validRegex = "^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$";
        Assert.isTrue(value.matches(validRegex),
                String.format("Email must be valid, but was %s", value));
    }
}

File: src/main/kotlin/com/manageway/domain/user/Email.kt
package com.manageway.domain.user

data class Email(
    val value: String
) {
    init {
        require(value.length in 5..60) { "Email must be between 5 and 60 characters" }
        require(value.matches(Regex("^[A-Za-z0-9+_.-]+@(.+)\$"))) { "Email must be valid" }
    }
}